### PR TITLE
add internal transactions into event structure

### PR DESCRIPTION
**Pull request by Maxime2**
and rename payload into transactions.

### CODE REVIEW

File: src/event.rs [hunk 1]
```diff
@@ -258,9 +266,9 @@ proptest! {
     #[test]
     fn it_should_have_different_hashes_on_different_transactions(tx1 in "[a-z]*", tx2 in "[a-z]*") {
         use crate::event::parents::ParentsPair;
-        let event1: Event<ParentsPair> = Event::new(vec![tx1.as_bytes().to_vec()], None, Vec::new());
-        let event2: Event<ParentsPair> = Event::new(vec![tx2.as_bytes().to_vec()], None, Vec::new());
-        let event3: Event<ParentsPair> = Event::new(vec![tx2.as_bytes().to_vec()], None, Vec::new());
+        let event1: Event<ParentsPair> = Event::new(vec![tx1.as_bytes().to_vec()], Vec::new(), None, Vec::new());
+        let event2: Event<ParentsPair> = Event::new(vec![tx2.as_bytes().to_vec()], Vec::new(), None, Vec::new());
+        let event3: Event<ParentsPair> = Event::new(vec![tx2.as_bytes().to_vec()], Vec::new(), None, Vec::new());
```
Comment (SamuelMarks) [inline src/event.rs]: Can you change them all to `vec![]` XOR `Vec::new()`, the lack of consistency gets to me 😅

Comment (Maxime2) [reply src/event.rs]: Changed to vec![] - more compact :)

Comment (SamuelMarks) [reply src/event.rs]: Great, there's still plenty missing though. Try a:
```
fd -t f -e rs -exec sed -i 's/Vec::new()/vec![]/g' {} \;
```

Comment (Maxime2) [reply src/event.rs]: changed it globally :)

**Comment by SamuelMarks** [comment]
Happy to merge, just a couple of conflicts which need resolving

**Comment by Maxime2** [comment]
@SamuelMarks : resolved.

**Pull request merged**